### PR TITLE
pdutil(ticdc): split `tidb_ddl_job` table

### LINKED ISSUE #4756: Meta region merge may cause latency spike
image

```log
[2022/03/02 03:21:22.530 +08:00] [INFO] [apply.rs:2495] ["execute CommitMerge"] [source_region="id: 33380 end_key: 7480000000000000FF175F728000000000FF000F3C0000000000FA region_epoch { conf_ver: 6 version: 115 } peers { id: 33381 store_id: 1 } peers { id: 33382 store_id:
[2022/03/02 03:21:22.530 +08:00] [INFO] [peer.rs:2964] ["notify pd with merge"] [target_region="id: 33548 end_key: 7480000000000000FF175F728000000000FF000F7E0000000000FA region_epoch { conf_ver: 5 version: 116 } peers { id: 33549 store_id: 1 } peers { id: 33550 store_id:
[2022/03/02 03:35:13.765 +08:00] [INFO] [apply.rs:2495] ["execute CommitMerge"] [source_region="id: 33548 end_key: 7480000000000000FF175F728000000000FF000F7E0000000000FA region_epoch { conf_ver: 6 version: 117 } peers { id: 33549 store_id: 1 } peers { id: 33550 store_id:
[2022/03/02 03:35:13.766 +08:00] [INFO] [peer.rs:2964] ["notify pd with merge"] [target_region="id: 33480 end_key: 7480000000000000FF175F728000000000FF000FA50000000000FA region_epoch { conf_ver: 5 version: 118 } peers { id: 33481 store_id: 1 } peers { id: 33482 store_id:
[2022/03/02 03:37:30.519 +08:00] [INFO] [apply.rs:2495] ["execute CommitMerge"] [source_region="id: 33480 end_key: 7480000000000000FF175F728000000000FF000FA50000000000FA region_epoch { conf_ver: 6 version: 119 } peers { id: 33481 store_id: 1 } peers { id: 33482 store_id:
[2022/03/02 03:37:30.519 +08:00] [INFO] [peer.rs:2964] ["notify pd with merge"] [target_region="id: 33424 end_key: 7480000000000000FF175F728000000000FF000FDE0000000000FA region_epoch { conf_ver: 5 version: 120 } peers { id: 33425 store_id: 1 } peers { id: 33426 store_id:
[2022/03/02 04:12:06.786 +08:00] [INFO] [apply.rs:2495] ["execute CommitMerge"] [source_region="id: 33424 end_key: 7480000000000000FF175F728000000000FF000FDE0000000000FA region_epoch { conf_ver: 6 version: 121 } peers { id: 33425 store_id: 1 } peers { id: 33426 store_id:
[2022/03/02 04:38:00.557 +08:00] [INFO] [apply.rs:2495] ["execute CommitMerge"] [source_region="id: 33744 end_key: 7480000000000000FF175F728000000000FF0010270000000000FA region_epoch { conf_ver: 6 version: 123 } peers { id: 33745 store_id: 1 } peers { id: 33746 store_id:
[2022/03/02 04:40:03.330 +08:00] [INFO] [apply.rs:2495] ["execute CommitMerge"] [source_region="id: 33484 end_key: 7480000000000000FF175F728000000000FF0010500000000000FA region_epoch { conf_ver: 6 version: 125 } peers { id: 33485 store_id: 1 } peers { id: 33486 store_id:
[2022/03/02 04:40:38.384 +08:00] [INFO] [apply.rs:1370] ["execute admin command"] [command="cmd_type: PrepareMerge prepare_merge { min_index: 13454 target { id: 33736 end_key: 7480000000000000FF175F728000000000FF00107E0000000000FA region_epoch { conf_ver: 5 version: 126 }
[2022/03/02 04:40:38.394 +08:00] [INFO] [peer.rs:3080] ["merge finished"] [target_region="Some(id: 33736 end_key: 7480000000000000FF175F728000000000FF00107E0000000000FA region_epoch { conf_ver: 5 version: 126 } peers { id: 33737 store_id: 1 } peers { id: 33738 store_id: 5
[2022/03/02 04:40:41.902 +08:00] [INFO] [apply.rs:2495] ["execute CommitMerge"] [source_region="id: 33736 end_key: 7480000000000000FF175F728000000000FF00109B0000000000FA region_epoch { conf_ver: 6 version: 128 } peers { id: 33737 store_id: 1 } peers { id: 33738 store_id:
[2022/03/02 04:40:45.392 +08:00] [INFO] [apply.rs:2495] ["execute CommitMerge"] [source_region="id: 33732 end_key: 7480000000000000FF175F728000000000FF0010C90000000000FA region_epoch { conf_ver: 6 version: 130 } peers { id: 33733 store_id: 1 } peers { id: 33734 store_id:
[2022/03/02 04:41:20.468 +08:00] [INFO] [apply.rs:1370] ["execute admin command"] [command="cmd_type: PrepareMerge prepare_merge { min_index: 14040 target { id: 33684 end_key: 7480000000000000FF175F728000000000FF0010E50000000000FA region_epoch { conf_ver: 5 version: 131 }
[2022/03/02 04:41:20.479 +08:00] [INFO] [peer.rs:3080] ["merge finished"] [target_region="Some(id: 33684 end_key: 7480000000000000FF175F728000000000FF0010E50000000000FA region_epoch { conf_ver: 5 version: 131 } peers { id: 33685 store_id: 1 } peers { id: 33686 store_id: 5
```

TiCDC lag increases when meta region gets merged.

### PR DESCRIPTION
**Pull request by overvenus**
### What problem does this PR solve?

Issue Number: close #4756

### What is changed and how it works?

Split the table `tidb_ddl_job`.

#### Tests 

 - Unit test
 - Manual test

Logs from PD leader.

```log
[2022/08/09 08:33:44.514 +00:00] [INFO] [operator_controller.go:450] ["add operator"] [region-id=8978001] [operator="\"labeler-split-region {split: region 8978001 use policy USEKEY and keys [748000FFFFFFFFFFFFFE00000000000000F8 748000FFFFFFFFFFFFFF00000000000000F8]} (kind:split, region:8978001(4850, 5), createAt:2022-08-09 08:33:44.512985722 +0000 UTC m=+1131983.134539904, startAt:0001-01-01 00:00:00 +0000 UTC, currentStep:0, size:2, steps:[split region with policy USEKEY])\""] [additional-info="{\"region-end-key\":\"\",\"region-start-key\":\"7480000000000003FF5F5F728000000000FF436EF30000000000FA\"}"]
[2022/08/09 08:33:44.516 +00:00] [INFO] [operator_controller.go:652] ["send schedule command"] [region-id=8978001] [step="split region with policy USEKEY"] [source=create]
[2022/08/09 08:33:44.519 +00:00] [INFO] [cluster_worker.go:139] ["alloc ids for region split"] [region-id=10365001] [peer-ids="[10365002,10365003,10365004]"]
[2022/08/09 08:33:44.519 +00:00] [INFO] [cluster_worker.go:139] ["alloc ids for region split"] [region-id=10365005] [peer-ids="[10365006,10365007,10365008]"]
[2022/08/09 08:33:44.524 +00:00] [INFO] [region.go:580] ["region Version changed"] [region-id=8978001] [detail="StartKey Changed:{7480000000000003FF5F5F728000000000FF436EF30000000000FA} -> {748000FFFFFFFFFFFFFF00000000000000F8}, EndKey:{}"] [old-version=4850] [new-version=4852]
[2022/08/09 08:33:44.527 +00:00] [INFO] [operator_controller.go:566] ["operator finish"] [region-id=8978001] [takes=11.372527ms] [operator="\"labeler-split-region {split: region 8978001 use policy USEKEY and keys [748000FFFFFFFFFFFFFE00000000000000F8 748000FFFFFFFFFFFFFF00000000000000F8]} (kind:split, region:8978001(4850, 5), createAt:2022-08-09 08:33:44.512985722 +0000 UTC m=+1131983.134539904, startAt:2022-08-09 08:33:44.516296175 +0000 UTC m=+1131983.137850403, currentStep:1, size:2, steps:[split region with policy USEKEY]) finished\""] [additional-info="{\"region-end-key\":\"\",\"region-start-key\":\"7480000000000003FF5F5F728000000000FF436EF30000000000FA\"}"]
[2022/08/09 08:33:44.526 +00:00] [INFO] [cluster_worker.go:231] ["region batch split, generate new regions"] [region-id=8978001] [origin="id:10365001 start_key:\"7480000000000003FF5F5F728000000000FF436EF30000000000FA\" end_key:\"748000FFFFFFFFFFFFFE00000000000000F8\" region_epoch:<conf_ver:5 version:4852 > peers:<id:10365002 store_id:1 > peers:<id:10365003 store_id:4 > peers:<id:10365004 store_id:5 > id:10365005 start_key:\"748000FFFFFFFFFFFFFE00000000000000F8\" end_key:\"748000FFFFFFFFFFFFFF00000000000000F8\" region_epoch:<conf_ver:5 version:4852 > peers:<id:10365006 store_id:1 > peers:<id:10365007 store_id:4 > peers:<id:10365008 store_id:5 >"] [total=2]
```

```shell
$ pd-ctl region 10365001
{
  "id": 10365001,
  "start_key": "7480000000000003FF5F5F728000000000FF436EF30000000000FA",
  "end_key": "748000FFFFFFFFFFFFFE00000000000000F8",
  "epoch": {
    "conf_ver": 5,
    "version": 4852
  },
  "peers": [
    {
      "id": 10365002,
      "store_id": 1,
      "role_name": "Voter"
    },
    {
      "id": 10365003,
      "store_id": 4,
      "role_name": "Voter"
    },
    {
      "id": 10365004,
      "store_id": 5,
      "role_name": "Voter"
    }
  ],
  "leader": {
    "id": 10365002,
    "store_id": 1,
    "role_name": "Voter"
  },
  "written_bytes": 0,
  "read_bytes": 0,
  "written_keys": 0,
  "read_keys": 0,
  "approximate_size": 2,
  "approximate_keys": 10813
}

$ pd-ctl region 8978001
{
  "id": 8978001,
  "start_key": "748000FFFFFFFFFFFFFF00000000000000F8",
  "end_key": "",
  "epoch": {
    "conf_ver": 5,
    "version": 4852
  },
  "peers": [
    {
      "id": 8978002,
      "store_id": 1,
      "role_name": "Voter"
    },
    {
      "id": 8978003,
      "store_id": 4,
      "role_name": "Voter"
    },
    {
      "id": 8978004,
      "store_id": 5,
      "role_name": "Voter"
    }
  ],
  "leader": {
    "id": 8978002,
    "store_id": 1,
    "role_name": "Voter"
  },
  "written_bytes": 362,
  "read_bytes": 0,
  "written_keys": 6,
  "read_keys": 0,
  "approximate_size": 1,
  "approximate_keys": 0
}
```

##### Will it cause performance regression or break compatibility?

No.

##### Do you need to update user documentation, design documentation or monitoring documentation?

No.

### Release note 

```release-note
None
```

### CODE REVIEW

File: pkg/pdutil/pd_api_client_test.go [hunk 1]
```diff
@@ -95,3 +100,53 @@ func TestListGcServiceSafePoint(t *testing.T) {
...
+	Index    int           `json:"index"`
+	Labels   []RegionLabel `json:"labels"`
+	RuleType string        `json:"rule_type"`
+	Data     interface{}   `json:"data"`
+}
+
+// RegionLabel is the label of a region.
+// NOTE: This type is exported by HTTP API. Please pay more attention when modifying it.
+// Copied from github.com/tikv/pd/server/schedule/labeler
+type RegionLabel struct {
```
Comment (lonng) [inline pkg/pdutil/pd_api_client_test.go:126]: Why not import the `github.com/tikv/pd/server/schedule/labeler` directly? I saw it's a public type.

See: https://github.com/tikv/pd/blob/master/server/schedule/labeler/rules.go#L32

**Review by lonng — COMMENTED** [review]
Rest LGTM

File: pkg/pdutil/pd_api_client_test.go (hunk 1 above)
Comment (overvenus) [reply pkg/pdutil/pd_api_client_test.go:126]: It make TiCDC depends on pd server which may bloat TiCDC binary size and dependency graph.

Comment (lonng) [reply pkg/pdutil/pd_api_client_test.go:126]: Actually, we have depends on it. see: https://github.com/pingcap/tiflow/blob/master/go.mod#L77

**Review by CharlesCheung96 — APPROVED** [review]

**Comment by lance6716** [comment]
there are 3 tables newly introduced in 6.2

https://github.com/pingcap/tidb/blob/28c934cdae80d15f1433ad5948bbc46885532b8e/ddl/constant.go#L21-L27

File: pkg/pdutil/pd_api_client_test.go (hunk 1 above)
Comment (overvenus) [reply pkg/pdutil/pd_api_client_test.go:126]: Tried, failed to compile tests
```shell
make tidy     
go mod tidy
./tools/check/check-tidy.sh
go: finding module for package google.golang.org/grpc/naming
github.com/pingcap/tiflow/pkg/pdutil tested by
        github.com/pingcap/tiflow/pkg/pdutil.test imports
        github.com/tikv/pd/server/schedule/labeler imports
        github.com/tikv/pd/server/storage/endpoint imports
        go.etcd.io/etcd/clientv3 tested by
        go.etcd.io/etcd/clientv3.test imports
        go.etcd.io/etcd/integration imports
        go.etcd.io/etcd/proxy/grpcproxy imports
        google.golang.org/grpc/naming: module google.golang.org/grpc@latest found (v1.48.0), but does not contain package google.golang.org/grpc/naming
make: *** [Makefile:292: tidy] Error 1
```

We have to upgrade etcd version in PD or patch etcd in TiCDC side, I think it's not worth to do so, only complicates already complexed dependency.

**Comment by overvenus** [comment]
> there are 3 tables newly introduced in 6.2
> 
> https://github.com/pingcap/tidb/blob/28c934cdae80d15f1433ad5948bbc46885532b8e/ddl/constant.go#L21-L27

TiCDC cares `tidb_ddl_job` only.

**Review by 3AceShowHand — APPROVED** [review]

File: pkg/pdutil/pd_api_client_test.go (hunk 1 above)
Comment (lonng) [reply pkg/pdutil/pd_api_client_test.go:126]: Sadly

**Review by lonng — APPROVED** [review]

**Pull request merged**